.github/workflows/build-on-pull-request.yml

```
@@ -0,0 +1,22 @@
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+
+      - name: Setup JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: 17
+          distribution: 'adopt'
+      - name: Build with Maven  
```

yamllint

[error] trailing spaces

(trailing-spaces)